### PR TITLE
xDD-172: Cosmos service docs

### PR DESCRIPTION
**Pull request by mwestphall**
- Update the route that FastAPI documentation is hosted at so that it's accessible from the k8s ingress path prefix
- Replace python dicts with data classes so that FastAPI's doc auto-generation feature picks up on them

### CODE REVIEW

File: cosmos_service/src/app.py [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
+from model.models import *
 import shutil
 
-app = FastAPI()
+prefix_url = os.environ.get('API_PREFIX','/')
+
+app = FastAPI(title="COSMOS Service", docs_url=f"{prefix_url}/docs")
 
-prefix_router = APIRouter(prefix=os.environ.get('API_PREFIX',''))
+prefix_router = APIRouter(prefix=prefix_url)
```
Comment (iross) [inline cosmos_service/src/app.py:22]: I think this is an invalid prefix in the case of the default `/` for prefix_url:

```
Traceback (most recent call last):
  File "/usr/local/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/main.py", line 416, in main
    run(
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/main.py", line 587, in run
    server.run()
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/server.py", line 68, in serve
    config.load()
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/config.py", line 467, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/src/app.py", line 22, in <module>
    prefix_router = APIRouter(prefix=prefix_url)
  File "/usr/local/lib/python3.8/dist-packages/fastapi/routing.py", line 522, in __init__
    assert not prefix.endswith(
AssertionError: A path prefix must not end with '/', as the routes will start with '/'
```

**Review by iross — COMMENTED** [review]
Looks good once the above is fixed!

File: cosmos_service/src/app.py [hunk 2]
```diff
@@ -12,12 +12,14 @@
 from db.db import SessionLocal, get_job_details, get_cached_job_for_pdf
 from scheduler import scheduler
 from util.cosmos_output_utils import extract_file_from_job, convert_parquet_to_json, replace_url_suffix
-
+from model.models import *
 import shutil
 
-app = FastAPI()
+prefix_url = os.environ.get('API_PREFIX','/')
```
Comment (mwestphall) [inline cosmos_service/src/app.py:18]:
```suggestion
prefix_url = os.environ.get('API_PREFIX','/cosmos_service')
```

**Review by iross — APPROVED** [review]
👍

**Pull request merged**